Makefile

```diff
@@ -12,94 +14,84 @@ else
+$(BUILD_DIR)/$(JAG_BINARY): $(JAG_GO_SOURCES) $(BUILD_DIR)
 	$(GO_BUILD_FLAGS) go build -tags 'netgo osusergo' -ldflags "$(GO_LINK_FLAGS)" -o $@ ./cmd/jag
 
-$(BUILD_DIR)/macos:
-	mkdir -p $@
-
 .PHONY: jag-macos-sign
 jag-macos-sign:
 	gon -log-level=debug -log-json ./tools/gon.json
 
-.PHONY: toit-git-tags
-toit-git-tags:
-	(cd $(TOIT_PATH); git fetch --tags --recurse-submodules=no)
-
-.PHONY: $(JAG_TOIT_PATH)/bin/toit.compile $(JAG_TOIT_PATH)/bin/toit.pkg
-$(JAG_TOIT_PATH)/bin/toit.compile $(JAG_TOIT_PATH)/bin/toit.pkg: toit-git-tags
-	make -C $(TOIT_PATH) sdk
+.PHONY: assets
+assets: $(BUILD_DIR)/assets/jaguar.snapshot
 
-.PHONY: $(TOIT_PATH)/build/esp32/firmware.envelope
-$(TOIT_PATH)/build/esp32/firmware.envelope: toit-git-tags
-	make -C $(TOIT_PATH) esp32
+$(BUILD_DIR)/assets/jaguar.snapshot: install-dependencies
+$(BUILD_DIR)/assets/jaguar.snapshot: $(JAG_TOIT_SOURCES)
+$(BUILD_DIR)/assets/jaguar.snapshot: $(SDK_PATH)/bin/toit.compile $(BUILD_DIR)/assets/
+	$(SDK_PATH)/bin/toit.compile -w $@ $(JAG_ENTRY_POINT)
 
 $(BUILD_DIR)/assets/:
 	mkdir -p $@
 
-$(BUILD_DIR)/assets/firmware.envelope: $(TOIT_PATH)/build/esp32/firmware.envelope $(BUILD_DIR)/assets/
-	cp $< $@
+.PHONY: install-dependencies
+install-dependencies: $(SDK_PATH)/bin/toit.pkg
+	$(SDK_PATH)/bin/toit.pkg --project-root=$(CURDIR) install
 
-$(BUILD_DIR)/assets/jaguar.snapshot: install-dependencies
-$(BUILD_DIR)/assets/jaguar.snapshot: $(TOIT_SOURCE)
-$(BUILD_DIR)/assets/jaguar.snapshot: $(JAG_TOIT_PATH)/bin/toit.compile $(BUILD_DIR)/assets/
-	$(JAG_TOIT_PATH)/bin/toit.compile -w $@ $(JAG_ENTRY_POINT)
+clean:
+	rm -rf $(BUILD_DIR)
 
-.PHONY: assets
-assets: $(BUILD_DIR)/assets/jaguar.snapshot
-assets: $(BUILD_DIR)/assets/firmware.envelope
+###############################################
+# Rules to build without JAG_TOIT_REPO_PATH set
+###############################################
 
-.PHONY: install-esp-idf
-install-esp-idf:
-	IDF_PATH=$(IDF_PATH) $(IDF_PATH)/install.sh
+.PHONY: download-sdk
+download-sdk: $(BUILD_DIR)/$(JAG_BINARY)
+	rm -rf $(BUILD_SDK_DIR)
+	$(BUILD_DIR)/$(JAG_BINARY) --no-analytics setup sdk $(BUILD_SDK_DIR)
 
-.PHONY: install-dependencies
-install-dependencies: $(JAG_TOIT_PATH)/bin/toit.pkg
-	$(JAG_TOIT_PATH)/bin/toit.pkg --project-root=$(CURDIR) install
+############################################
+# Rules to build with JAG_TOIT_REPO_PATH set
+############################################
 
-clean:
-	rm -rf $(BUILD_DIR)
+ifdef JAG_TOIT_REPO_PATH
+all: $(JAG_TOIT_REPO_PATH)/build/esp32/firmware.envelope
+
+JAG_TOIT_DEPENDENCIES  = $(SDK_PATH)/bin/toit.compile
+JAG_TOIT_DEPENDENCIES += $(SDK_PATH)/bin/toit.pkg
+JAG_TOIT_DEPENDENCIES += $(JAG_TOIT_REPO_PATH)/build/esp32/firmware.envelope
+
+.PHONY: $(JAG_TOIT_DEPENDENCIES)
+$(JAG_TOIT_DEPENDENCIES): toit-repo-git-tags
+	make -C $(JAG_TOIT_REPO_PATH) version-file esp32
+
+.PHONY: toit-repo-git-tags
+toit-repo-git-tags:
+	(cd $(JAG_TOIT_REPO_PATH); git fetch --tags --recurse-submodules=no)
```

Comment:
I would remove that one.
When we work with a locally checked out toit-repo, I expect that directory to have all submodules checked out, and all tags to be there.

Reply:
Done.